packages/shared/src/hooks/useBanner.ts

```diff
@@ -0,0 +1,58 @@
+import { useContext, useMemo } from 'react';
+import { useMutation, useQuery } from 'react-query';
+import request from 'graphql-request';
+import AlertContext from '../contexts/AlertContext';
+import { generateQueryKey, RequestKey } from '../lib/query';
+import { graphqlUrl } from '../lib/config';
+import { Banner, BANNER_QUERY } from '../graphql/banner';
+
+type UseBanner = {
+  isAvailable: boolean;
+  latestBanner?: Banner;
+  dismiss: () => Promise<void>;
+};
+
+export function useBanner(): UseBanner {
+  const { alerts, updateAlerts } = useContext(AlertContext);
+
+  const { data: latestBanner } = useQuery(
+    generateQueryKey(RequestKey.Banner, null),
+    () => request(graphqlUrl, BANNER_QUERY, { lastSeen: alerts.lastBanner }),
+    {
+      enabled: !!alerts.banner,
+    },
+  );
+
+  const isAvailable = useMemo(() => {
+    const lastBannerDate = Date.parse(alerts?.lastBanner);
+    const lastPostDate = Date.parse(latestBanner?.banner?.timestamp);
+
+    if (Number.isNaN(lastBannerDate) || Number.isNaN(lastPostDate)) {
+      return false;
+    }
+
+    return lastPostDate > lastBannerDate;
+  }, [alerts.lastBanner, latestBanner?.banner?.timestamp]);
```

Comment:
Not sure I understand this part

Reply:
Ill refactor and make it more clear 🙏

Reply:
@idoshamun Refactored a bit, it's basically the same "extra" evaluation we use for useChangelog.
To ensure you should actually see the banner based on when you last seen it.

The main idea was I think that boot could potentially get lagged behind if you didn't particularly refreshed.